src/main/java/mate/academy/spring/dao/impl/RentDaoImpl.java

```diff
@@ -0,0 +1,45 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.persistence.TypedQuery;
+import mate.academy.spring.dao.RentDao;
+import mate.academy.spring.entity.Book;
+import mate.academy.spring.entity.Rent;
+import mate.academy.spring.entity.User;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RentDaoImpl implements RentDao {
+    @Autowired
+    private SessionFactory sessionFactory;
+
+    @Override
+    public Rent rentBook(Rent rent) {
+        Long id = (Long) sessionFactory.getCurrentSession().save(rent);
+        rent.setId(id);
+        return rent;
+    }
+
+    @Override
+    public void returnBook(User user, Book book) {
+        TypedQuery<Rent> query = sessionFactory.getCurrentSession()
+                .createQuery("FROM Rent WHERE user = :user AND book = :book", Rent.class);
+        query.setParameter("user", user);
+        query.setParameter("book", book);
+        Rent rent = query.getSingleResult();
+        rent.setActive(false);
+        sessionFactory.getCurrentSession().update(rent);
+    }
+
+    @Override
+    public List<Book> getBooksRentByUser(User user) {
+        TypedQuery<Rent> query = sessionFactory.getCurrentSession()
+                .createQuery("FROM Rent WHERE user = :user AND active = true", Rent.class);
```

Comment:
```suggestion
                .createQuery("SELECT rent.book FROM Rent WHERE user = :user AND active = true", Rent.class);
```
after that try removing line with stream()

Reply:
Предложенный вариант не работает. Хотел бы я знать, как такой запрос написать с помощью HQL или как-нибудь в объектном стиле. На чистом SQL запрос будет выглядеть так: "SELECT books.book_id, books.price, books.title, books.year FROM rent INNER JOIN books ON rent.book_id=books.book_id WHERE rent.user_id = ?;" Хотелось бы подробнее, как правильно в hibernate делать запросы на выборку из двух таблиц

Reply:
https://stackoverflow.com/questions/15068597/inner-joins-query-in-hql
https://www.concretepage.com/hibernate/hibernate-hql-associations-and-inner-join-left-outer-join-right-outer-join-cross-join-example

Reply:
Разобрался: в твоём предложенном запросе нужно было ещё указать алиас "FROM Rent AS rent", иначе хибернейт не понимает "SELECT rent.book". И ещё тип класса в конце поменять на Book.class.
SELECT rent.book FROM Rent AS rent WHERE rent.user = :user AND active = true", Book.class